packages/perseus/src/widgets/numeric-input/numeric-input.class.tsx

```diff
@@ -157,79 +156,34 @@ export class NumericInput
-    // Takes a list of lists of answer forms, and returns a list of the forms
-    // in each of these lists in the same order that they're listed in the
-    // `formExamples` forms from above.
-
-    // uniqueBy takes a list of elements and a function which compares whether
-    // two elements are equal, and returns a list of unique elements. This is
-    // just a helper function here, but works generally.
-    const uniqueBy = function (list, iteratee: any) {
-        // @ts-expect-error - TS2347 - Untyped function calls may not accept type arguments.
-        return list.reduce<Array<any>>((uniqueList, element) => {
-            // For each element, decide whether it's already in the list of
-            // unique items.
-            const inList = _.find(uniqueList, iteratee.bind(null, element));
-            if (inList) {
-                return uniqueList;
-            }
-            return uniqueList.concat([element]);
-        }, []);
-    };
-
-    // Pull out all of the forms from the different lists.
-    const allForms = answerFormsList.flat();
-    // Pull out the unique forms using uniqueBy.
-    const uniqueForms = uniqueBy(allForms, _.isEqual);
-    // Sort them by the order they appear in the `formExamples` list.
-    const formExampleKeys = Object.keys(NumericExampleStrings);
-    return _.sortBy(uniqueForms, (form) => {
-        return formExampleKeys.indexOf(form.name);
-    });
-};
-
 type RenderProps = {
-    answerForms: ReadonlyArray<{
-        simplify: "required" | "correct" | "enforced" | null | undefined;
-        name: "integer" | "decimal" | "proper" | "improper" | "mixed" | "pi";
-    }>;
-    labelText: string;
-    size: "normal" | "small";
+    answerForms: ReadonlyArray<PerseusNumericInputAnswerForm>;
+    labelText?: string;
+    size: string;
     coefficient: boolean;
     rightAlign?: boolean;
     static: boolean;
 };
 
+// Transforms the widget options to the props used to render the widget.
 const propsTransform = function (
     widgetOptions: PerseusNumericInputWidgetOptions,
 ): RenderProps {
-    const rendererProps = _.extend(_.omit(widgetOptions, "answers"), {
+    const rendererProps: RenderProps & {answers?: any} = {
+        ...widgetOptions,
         answerForms: unionAnswerForms(
-            // Pull out the name of each form and whether that form has
-            // required simplification.
             widgetOptions.answers.map((answer) => {
-                // @ts-expect-error - TS2345 - Argument of type 'readonly MathFormat[] | undefined' is not assignable to parameter of type 'Collection<any>'.
-                return _.map(answer.answerForms, (form) => {
+                return (answer.answerForms || []).map((form) => {
                     return {
                         simplify: answer.simplify,
                         name: form,
                     };
                 });
             }),
         ),
-    });
+    };
 
+    delete rendererProps.answers;
```

Comment:
Instead of needing to jump through this typing hoop, what do you think of restructuring the `answers` away? 

```typescript
   const {answers: _, ...rendererProps} = {
         ...widgetOptions,
         answerForms: unionAnswerForms(
             widgetOptions.answers.map((answer) => {
                 return (answer.answerForms || []).map((form) => {
                     return {
                         simplify: answer.simplify,
                         name: form,
                     };
                 });
             }),
         ),
     };
```

Reply:
Much better, thank you!